static/panes/executor.js

```diff
@@ -122,7 +122,7 @@ function Executor(hub, container, state) {
     // Handle initial settings
     this.onSettingsChange(this.settings);
     this.updateCompilerInfo();
-    this.saveState();
+    this.compile();
```

Comment:
cool! I wonder why this is not needed in other panes...gonna take a look to refresh my memory!

Reply:
Ah...the compiler gets an onEditorChange callback on the initial contents of the editor window. I wonder why the execution pane doesn't get this?

Reply:
@RabsRincon I think your change 20bc613 removed a function that's used: I just synced and got:

```
jquery.js:3850 Uncaught TypeError: this.initPanerButtons is not a function
    at Executor.initButtons (executor.js:437)
    at new Executor (executor.js:83)
```

Reply:
So...I _think_ the executor should not do a `compile` here: depending on the order the windows are stored in the serialised format the editor may not yet have sent its window contents.

The horrible code to handle this is in compiler.js: the editor window resends a `requestCompilation` message (in `Editor.prototype.requestCompilation`) once it has been initialised. I _think_ that's all that's needed here.

I tested this locally (hackily) and seems to work:

```diff
diff --git a/static/panes/executor.js b/static/panes/executor.js
index 692c79a0..f8b4685b 100644
--- a/static/panes/executor.js
+++ b/static/panes/executor.js
@@ -122,7 +122,8 @@ function Executor(hub, container, state) {
     // Handle initial settings
     this.onSettingsChange(this.settings);
     this.updateCompilerInfo();
-    this.compile();
+    // this.compile();
+    this.saveState();
     ga.proxy('send', {
         hitType: 'event',
         eventCategory: 'OpenViewPane',
@@ -290,7 +291,7 @@ Executor.prototype.addCompilerOutputLine = function (msg, container, lineNum, co
                     return false;
                 }, this))
                 .on('mouseover', _.bind(function () {
-                    this.eventHub.emit('editorLinkLine', this.sourceEditorId, lineNum,  column, false);
+                    this.eventHub.emit('editorLinkLine', this.sourceEditorId, lineNum, column, false);
                 }, this))
         );
     } else {
@@ -434,7 +435,7 @@ Executor.prototype.initButtons = function (state) {
     this.hideable = this.domRoot.find('.hideable');
     this.statusIcon = this.domRoot.find('.status-icon');
 
-    this.initPanerButtons();
+    // this.initPanerButtons();
 };
 
 Executor.prototype.onLibsChanged = function () {
@@ -458,6 +459,7 @@ Executor.prototype.initListeners = function () {
     this.container.on('destroy', this.close, this);
     this.container.on('resize', this.resize, this);
     this.container.on('shown', this.resize, this);
+    this.eventHub.on('requestCompilation', this.onRequestCompilation, this);
     this.container.on('open', function () {
         this.eventHub.emit('compilerOpen', this.id, this.sourceEditorId);
     }, this);
@@ -470,6 +472,12 @@ Executor.prototype.initListeners = function () {
     this.eventHub.on('languageChange', this.onLanguageChange, this);
 };
 
+Executor.prototype.onRequestCompilation = function (editorId) {
+    if (editorId === this.sourceEditorId) {
+        this.compile();
+    }
+};
+
 Executor.prototype.initCallbacks = function () {
     this.initListeners();
```